pkg/apis/certmanager/v1alpha2/const.go

```diff
@@ -38,5 +38,5 @@ const (
 
 	// Default mount path location for Kubernetes ServiceAccount authentication
 	// (/v1/auth/kubernetes/login)
-	DefaultVaultKubernetesAuthMountPath = "kubernetes"
+	DefaultVaultKubernetesAuthMountPath = "/v1/auth/kubernetes/login"
```

Comment:
Do we want to hardcode the `login` part of this? From what I understand, the `/v1/auth/kubernetes` path is the *mount point* of the kubernetes auth backend - I don't know if we _ever_ need (or even want) users to override the suffix. cc @mam8270 have you got any thoughts on this?

Reply:
Login can be a bit more nuanced now with the vault enterprise "namespaces" feature, which essentially allows vault inside vault for the different namespaces (including auth endpoints), so i think you would want at least some mechanism for overriding of the login url.  I agree you would lose some safety though.
https://learn.hashicorp.com/vault/operations/namespaces#policy-with-namespaces

Reply:
What's the conclusion here? If the path is wrong, a user can always override the path completely which is the point of this PR.

Reply:
Thanks for the link @mam8270 - I'm still a bit unclear on this though, the link you reference looks like it refers to *prefixes* on the Vault path, not the endpoint within the Kubernetes auth backend that is used to actually authenticate (i.e. `/login`).

Does this enterprise 'namespacing' feature allow this endpoint to be changed? If so, we will keep this line as is.

If not, we can make this field refer to the *mount point* of the auth backend instead of the particular endpoint used to authenticate (i.e. `/v1/auth/kubernetes` vs `/v1/auth/kubernetes/login`).

We want to allow people as much flexibility as possible, but if it isn't _possible_ to specify a value other than `login`, we can gain a bit of 'safety' and save users making configuration errors.

Reply:
To note, `mountPath` is the name of the field here.. which IMO implies `/v1/auth/kubernetes` *should* be a valid value, and `/v1/auth/kubernetes/login` *isn't* (as it is not hte *mount* path)

Reply:
We're going to go with `/v1/auth/kubernetes` for now due to the field being named `mountPath`. If we need more flexibility in future, we can introduce a `path` field instead.